otellog/otel_log.go

```diff
@@ -0,0 +1,70 @@
+package otellog
+
+import (
+	"fmt"
+	"io"
+	"sync"
+	"time"
+)
+
+type Logger interface {
+	Log(level LogLevel, msg fmt.Stringer)
+}
+
+type LogLevel int
+
+const (
+	// LogLevelDebug is usually only enabled when debugging.
+	LogLevelDebug LogLevel = iota + 1
```

Comment:
Should we try to align these with the levels currently defined in the [logging spec draft](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#severity-fields)?